bibblio.py

```diff
@@ -0,0 +1,391 @@
+        )
+        return credential.credential
+
+    @property
+    def default_headers(self):
+        return {
+            'Authorization': 'Bearer '+self.token,
+            'Content-Type': 'application/json'
+        }
+
+    def refresh_credential(self, credential):
+        url = self.API_ENDPOINT + 'token'
+        headers = {'Content-Type': self.TOKEN_CONTENT_TYPE}
+        client_details = dict(client_id=self.client_id, client_secret=self.client_secret)
+
+        response = HTTP.post_with_timeout(url, client_details, headers=headers)
+        data = response.json()
+
+        credential.credential = data.get('access_token')
+        expires_in = data.get('expires_in')
+        credential.expires = datetime.utcnow() + timedelta(0, expires_in * 0.9)
+        self._credential = credential
+
+    def create_catalogue(self, name, description=None):
+        catalogue = dict(name=name)
+        if description:
+            catalogue['description'] = description
+
+        catalogue = self.set_timestamp(catalogue, create=True)
+        catalogue = json.dumps(catalogue)
+
+        response = HTTP.post_with_timeout(
+            self.CATALOGUES_ENDPOINT, catalogue,
+            headers=self.default_headers,
+            allowed_response_codes=[201],
+            disallowed_response_codes=['4xx']
+        )
+
+        catalogue = response.json()
+
+        name = catalogue.get('name')
+        catalogue_id = catalogue.get('catalogueId')
+        self.log.info(
+            "New catalogue '%s' created with ID: %s", name, catalogue_id)
+
+        return catalogue
+
+    def get_catalogue(self, name):
+        response = HTTP.get_with_timeout(
+            self.CATALOGUES_ENDPOINT, headers=self.default_headers
+        )
+
+        if response.status_code == 200:
+            catalogues = response.json().get('results')
+            catalogue = filter(lambda c: c.get('name') == name, catalogues)
+            if catalogue:
+                return catalogue[0]
+            else:
+                return None
+
+    def create_content_item(self, content_item):
+        content_item = self.set_timestamp(content_item, create=True)
+        content_item = json.dumps(content_item)
+        response = HTTP.post_with_timeout(
+            self.CONTENT_ITEMS_ENDPOINT, content_item,
+            headers=self.default_headers,
+            allowed_response_codes=[201],
+            disallowed_response_codes=['4xx']
+        )
+
+        content_item = response.json()
+
+        name = content_item.get('name')
+        content_item_id = content_item.get('contentItemId')
+        self.log.info(
+            "New content item created for '%s': '%s'", name, content_item_id)
+
+        return content_item
+
+
+class BibblioCoverageProvider(WorkCoverageProvider):
+
+    DATA_SOURCE_NAME = DataSource.BIBBLIO
+    DEFAULT_BATCH_SIZE = 25
+    SERVICE_NAME = 'Bibblio Coverage Provider'
+    OPERATION = 'bibblio-export'
```

Comment:
There's no DataSource on `WorkCoverageRecords`, and I'm not sure it makes sense to add one for this integration juuuust yet.